sentry_sdk/tracing.py

```diff
@@ -543,24 +543,24 @@ def finish(self, hub=None):
         hub = hub or self.hub or sentry_sdk.Hub.current
         client = hub.client
 
+        if client is None:
+            # We have no client and therefore nowhere to send this transaction.
+            return None
+
         # This is a de facto proxy for checking if sampled = False
         if self._span_recorder is None:
             logger.debug("Discarding transaction because sampled = False")
 
             # This is not entirely accurate because discards here are not
             # exclusively based on sample rate but also traces sampler, but
             # we handle this the same here.
```

Comment:
Isn't this comment outdated now?

Reply:
hmm I think it tries to convey that we are recording drops due to both `traces_sample_rate` and `traces_sampler` which still holds.